### PR TITLE
New factory 4.0 model

### PR DESCRIPTION
**Pull request by mshaposhnik**

### CODE REVIEW

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/app/AppContext.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
  *******************************************************************************/
 package org.eclipse.che.ide.api.app;
 
-//import org.eclipse.che.api.factory.dto.Factory;
+//import org.eclipse.che.api.factory.shared.dto.Factory;
```
Comment (skabashnyuk) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/app/AppContext.java]: remove

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/bootstrap/FactoryComponent.java [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 package org.eclipse.che.ide.bootstrap;
 
-//import org.eclipse.che.api.factory.dto.Factory;
+//import org.eclipse.che.api.factory.shared.dto.Factory;
```
Comment (skabashnyuk) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/bootstrap/FactoryComponent.java]: remove

**Comment by skabashnyuk** [comment]
LGTM

File: platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java [hunk 3]
```diff
@@ -409,23 +380,24 @@ public Factory updateFactory(@ApiParam(value = "Factory ID", required = true)
...
-                                 .withProduces(MediaType.APPLICATION_JSON)
-                                 .withConsumes(null)
-                                 .withHref(UriBuilder.fromUri(uriInfo.getBaseUri())
-                                                     .path(FactoryService.class)
-                                                     .path(FactoryService.class, "getFactory").build(factory.getId())
-                                                     .toString())
-                                 .withParameters(null));
-        }
+        result.addAll(factories.stream()
+                               .map(factory -> DtoFactory.getInstance()
```
Comment (sleshchenko) [inline platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java]: You can use DtoFactory.newDto(... Also you can use static import for this.

**Pull request merged**